### ISSUE TITLE
Clean up and update read.output documentation 

### ISSUE THREAD

**Issue by serbinsh** [PecanProject/pecan#1952]
**Describe the bug**
The read.output documentation is considerably out-of-date and in need of a refresh to reflect the current functionality

Here is the current documentation:

```
##' Read model output
##'
##' Reads the output of a single model run
##'
##' Generic function to convert model output from model-specific format to 
##' a common PEcAn format. This function uses MsTMIP variables except that units of
##'  (kg m-2 d-1)  are converted to kg ha-1 y-1. Currently this function converts
##' Carbon fluxes: GPP, NPP, NEE, TotalResp, AutoResp, HeteroResp,
##' DOC_flux, Fire_flux, and Stem (Stem is specific to the BioCro model)
##' and Water fluxes: Evaporation (Evap), Transpiration(TVeg),
##' surface runoff (Qs), subsurface runoff (Qsb), and rainfall (Rainf).
##' For more details, see the
##' \href{http://nacp.ornl.gov/MsTMIP_variables.shtml}{MsTMIP variables}
##' documentation 
##' @param runid the id distiguishing the model run. 
##' @param outdir the directory that the model's output was sent to
##' @param start.year,end.year first and last year of output to read. Specify as a date-time (only the year portion is used) or as a four-digit number or string. If NA, reads all years found in \code{outdir}.
##' @param variables variables to be read from model output
##' @param dataframe Logical: if TRUE, will return output in a data.frame format with a posix column. Useful for align.data and plotting.
##' @param pft.name character string, name of the pft to read pft-specific output. If NULL no pft-specific output will be read even the variable has pft as a dimension.
##' @return If \code{dataframe=FALSE}, a vector of output variables. If \code{dataframe=TRUE}, a data.frame of output variables with POSIXct timestamps added.
##' @export
##' @author Michael Dietze, David LeBauer
```

We need to update to reflect the fact that read.output now reads already converted output to be used elsewhere.

Also, can we now fully deprecate:
```
##'--------------------------------------------------------------------------------------------------#
##' Converts the output of all model runs
##'
##' @title convert outputs from model specific code to pecan standard (deprecated) 
##' @name convert.outputs
##' @param model name of simulation model currently accepts ('ED', 'SIPNET', 'BIOCRO')
##' @param settings settings loaded from pecan.xml
##' @param ... arguments passed to \code{\link{read.output}}, e.g. \code{variables}, \code{start.year}, \code{end.year}
##' @export
##' @author Rob Kooper
convert.outputs <- function(model, settings, ...) {
  PEcAn.logger::logger.severe("DEPRECATED: This function is no longer used and will be removed in the future.")
} # convert.outputs

```
?

Finally, I am not sure we want to do this, do we?
```
  years <- start.year:end.year
  run_origin <- paste0(start.year, "-01-01") #-- should we be hard-coding this assumption?
```
That is, assume that the time is YYYY jan 1?  What about runs that start in the middle of the year or does this not matter since its the "origin"?

**Comment by serbinsh** [comment]
Oh, and another suggestion following in the vein of @dlebauer @ashiklom @infotroph, should we put one set of example model output (something very small, < 1Mb in size) in the package so we can run tests on read.output? Or did we decide not to include data like in the packages?

**Comment by infotroph** [comment]
> Finally, I am not sure we want to do this, do we?
>
>  ```
> years <- start.year:end.year
>   run_origin <- paste0(start.year, "-01-01") #-- should we be hard-coding this assumption?
>  ```
> That is, assume that the time is YYYY jan 1? What about runs that start in the middle of the year or does this not matter since its the "origin"?

It doesn't matter because it's the origin -- we just need to use *some* known time as our reference point, and having the origin be close to the start of the run is nice because it makes the raw value a little more interpretable than "seconds since 1582-10-15" or whatever. 

Additionally, since the output files have one-year granularity, it seemed reasonable to adjust the origin in one-year units -- in some sense a "2001.nc" that starts in July will always be missing data, even if it's missing on purpose :)

**Comment by infotroph** [comment]
> should we put one set of example model output (something very small, < 1Mb in size) in the package so we can run tests on read.output

Two points: 
1. We [already run tests](https://github.com/PecanProject/pecan/blob/develop/base/utils/tests/testthat/test-read.output.R) on read.output! For the current version decided it was more efficient to create mock outputs on the fly rather than store them, but in principle I'm not opposed.
2. It's worth thinking really hard about the size of any data we do commit. For perspective, CRAN wants the *entire package* to be less than 1 MB in all but exceptional circumstances. I know we're not CRAN, but it's a good reminder that for most tests we don't actually need a full-sized dataset.

**Comment by ashiklom** [comment]
@infotroph stole most of what I was about to say. I agree that mock- or code-generated outputs are better, but that it's better to air on the side of having something to test rather than having the test itself be super efficient.

But I also agree with point 2, that file size, should be something we think about, both here and more generally. I would add run time to that as well. Hadley's guide to testing R packages suggests that package unit tests shold take less than a minute to run.

**Comment by serbinsh** [comment]
I dont know how I missed the tests for read.output, I thought I took a peak for them?!   I must have looked in the wrong place, I was sort of juggling a few things this morning

**Comment by infotroph** [comment]
Would be good to resolve #562 at the same time as this, to make sure the documentation reflects any change in arguments

**Comment by infotroph** [comment]
I think @ashiklom addressed most of this in #2209, so closing. N.B. #562 remains open.
